packages/ingest/FtpProviderClient.js

```diff
@@ -74,38 +93,44 @@ class FtpProviderClient {
     const client = await this.buildFtpClient();
 
     return new Promise((resolve, reject) => {
-      client.on('error', reject);
+      client.on('error', this.errorHandler.bind(this, reject));
```

Comment:
Man I find the `.bind` syntax confusing. So the `reject` argument we're passing will *precede* the argument that would get passed in by default to the callback here, which is an error. 👌

Reply:
`.bind` is pretty simple if you think of it as a function where:
1. arg #1 is what the bound function's context will have as `this`.
2. args #2-infinity onwards are just pre-set into the args the function accepts. So `errorHandler.bind(this, reject);` just returns a function that is functionally identical to `errorHandler` but with `rejectFn` pre-set to `reject`, and not effectively in the arguments list. Which is why it only takes one argument, the error, like you said, even though that is argument 2 on the original function.

This gets harder with parameter objects. 🙂